### DISCUSSION TITLE
How to use cookie to choice i18n language for ssr

### DISCUSSION THREAD

**Discussion by zyxkad** [vikejs/vike#907] [Discussion] [answered]
Good morning/afternoon/evening everyone,

I want to render i18n on the server side, but I don't want to use hostname, url prefix or param such things, I only want to use cookie because it's really neat.

Before I move my project to ssr, I was using vue-i18n@9.2.2 and vue-cookies@1.8.3 to do this
But after I moved to SSR, I cannot use vue-cookies anymore cause `document.cookies` is not available on the server side.
I tried to pass the cookies that I received from express to the render function, but I have no way since render function is a hook

Any one can help me? Thanks in advance

**Comment by brillout** [comment, marked as answer]
You can pass the cookies at `renderPage()`, see docs.

**Reply by zyxkad** [reply]
Ah I see, I can pass things by the pageContext, but do you have any idea how to pass it when I do `vite`? it will start an internal server but not run my server.js

**Reply by brillout** [reply]
Use your own server instead of Vite's dev server. Vite's server is meant for pre-rendered apps, or very basic SSR apps.
